test/unit/specs/components/ActivityCollector/attachClickActivityCollector.spec.js

```diff
@@ -80,4 +79,15 @@ describe("ActivityCollector::attachClickActivityCollector", () => {
       expect(eventManager.sendEvent).not.toHaveBeenCalled();
     });
   });
+  it("returns undefined", () => {
+    eventManager.createEvent = () => {
+      return {
+        isEmpty: () => false
+      };
+    };
+    attachClickActivityCollector(cfg, eventManager, lifecycle);
+    clickHandler({}).then(result => {
```

Comment:
This promise needs to be returned, otherwise the test runner is not going to wait before moving on to the next test. It looks like other tests in the same spec file are suffering from the same problem. Would you please fix them as well while you're at it?

Reply:
thank you, I'll fix the other tests as well